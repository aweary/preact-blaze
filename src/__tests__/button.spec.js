import { h } from "preact";
import render from "preact-render-to-string";
import Button from "../button";

describe("Button", () => {
  it("exports a function", () => {
    expect(typeof Button).toBe("function");
  });

  it("renders a button with the base class name", () => {
    const rendered = render(<Button />);
    expect(rendered.includes("c-button")).toBe(true);
    expect(rendered).toMatchSnapshot();
  });

  it("renders the label prop as button text", () => {
    const rendered = render(<Button label="foo" />);
    expect(rendered.includes("foo")).toBe(true);
    expect(rendered).toMatchSnapshot();
  });

  it("renders the children as the button text if label is missing", () => {
    const rendered = render(<Button>bar</Button>);
    expect(rendered.includes("bar")).toBe(true);
    expect(rendered).toMatchSnapshot();
  });
  // Assert each variant renders the correct class
  [
    "brand",
    "info",
    "warning",
    "success",
    "error",
    "ghost",
    "block"
  ].forEach(variant => {
    it(`renders the ${variant} variant`, () => {
      const props = { [variant]: true };
      const rendered = render(<Button {...props} />);
      expect(rendered.includes(`c-button--${variant}`)).toBe(true);
      expect(rendered).toMatchSnapshot();
    });
  });

  ["xsmall", "small", "medium", "large", "xlarge", "super"].forEach(size => {
    it(`renders the ${size} size`, () => {
      const rendered = render(<Button size={size} />);
      expect(rendered.includes(`u-${size}`)).toBe(true);
      expect(rendered).toMatchSnapshot();
    });
  });
});
