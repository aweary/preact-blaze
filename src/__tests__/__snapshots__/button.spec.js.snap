exports[`Button renders a button with the base class name 1`] = `"<button class=\"c-button\"></button>"`;

exports[`Button renders the block variant 1`] = `"<button class="c-button c-button--block"></button>"`;

exports[`Button renders the brand variant 1`] = `"<button class="c-button c-button--brand"></button>"`;

exports[`Button renders the children as the button text if label is missing 1`] = `"<button class=\"c-button\">bar</button>"`;

exports[`Button renders the error variant 1`] = `"<button class="c-button c-button--error"></button>"`;

exports[`Button renders the ghost variant 1`] = `"<button class="c-button c-button--ghost"></button>"`;

exports[`Button renders the info variant 1`] = `"<button class="c-button c-button--info"></button>"`;

exports[`Button renders the label prop as button text 1`] = `"<button class=\"c-button\">foo</button>"`;

exports[`Button renders the large size 1`] = `"<button class="c-button u-large"></button>"`;

exports[`Button renders the medium size 1`] = `"<button class="c-button u-medium"></button>"`;

exports[`Button renders the small size 1`] = `"<button class="c-button u-small"></button>"`;

exports[`Button renders the success variant 1`] = `"<button class="c-button c-button--success"></button>"`;

exports[`Button renders the super size 1`] = `"<button class="c-button u-super"></button>"`;

exports[`Button renders the warning variant 1`] = `"<button class="c-button c-button--warning"></button>"`;

exports[`Button renders the xlarge size 1`] = `"<button class="c-button u-xlarge"></button>"`;

exports[`Button renders the xsmall size 1`] = `"<button class="c-button u-xsmall"></button>"`;
